src/dlr.cc

```diff
@@ -105,6 +106,77 @@ extern "C" int GetDLRNumOutputs(DLRModelHandle* handle, int* num_outputs) {
   API_END();
 }
 
+extern "C" int GetDLRHasMetadata(DLRModelHandle* handle, bool* has_metadata) {
+  API_BEGIN();
+  TVMModel* model = static_cast<TVMModel*>(*handle);
+  CHECK(model != nullptr) << "model is nullptr, create it first";
+  *has_metadata = model->HasMetadata();
+  API_END();
+}
+
+extern "C" int GetDLRInputNames(DLRModelHandle* handle, char*** names) {
+  API_BEGIN();
+  TVMModel* model = static_cast<TVMModel*>(*handle);
+  CHECK(model != nullptr) << "model is nullptr, create it first";
+  CHECK(model->HasMetadata()) << "model has no metadata";
```

Comment:
TFLite, TF backends do not need Metadata to get list of model InputNames. TVM backend does not need it as well

Reply:
@apivovarov there is a public `get_input` function in DLR which hardcodes input data type to float32 which isnt't right I guess . https://github.com/neo-ai/neo-ai-dlr/blob/master/python/dlr/dlr_model.py#L356. We can use the metadata file to set input type for that or delete the function `get_input` if its not required ?

Reply:
Python DLR `get_input` method will use DLR C API `GetDLRInputDataTypes` to get model inputs types instead of using hardcoded `float32`.
Internally `GetDLRInputDataTypes` will use `DLRModel::GetInputDataTypes`. DLRModel is abstract class, each impl (TF, TFLite, TVM) will use its own approach to impl `GetInputDataTypes`. `TensorflowModel` and `TFLiteModel` will use TF API and TFLite API to query `.pb` and `.tflite` files to get inputs name/type. `TVMModel` will use tvm graph_runtime functions - `GetInputType(index)` (just added) and `GetInputName(index)`.

Reply:
Discussed this offline. The nodes names are not consistent with uncompiled model node names. We should be reading output related data from metadata and input related data from graph_runtime.

Reply:
Lets read OutputType and InputType from model.json (`GetDLRInputType` and `GetDLROutputType` already added to dlr.h). Since  `model.json` does not have Original model OutputName we have to read it from metadata file. So, the only field that we need to read from metadata file is Original model OutputName.